src/integrationTest/kotlin/com/sourcegraph/cody/edit/DocumentCodeTest.kt

```diff
@@ -0,0 +1,152 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.util.Disposer
+import com.intellij.testFramework.EditorTestUtil
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import com.intellij.testFramework.runInEdtAndWait
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.intellij.util.messages.Topic
+import com.sourcegraph.cody.edit.widget.LensAction
+import com.sourcegraph.cody.edit.widget.LensGroupFactory
+import com.sourcegraph.cody.edit.widget.LensLabel
+import com.sourcegraph.cody.edit.widget.LensSpinner
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+class DocumentCodeTest : BasePlatformTestCase() {
+
+  override fun setUp() {
+    super.setUp()
+    configureFixture()
+  }
+
+  override fun tearDown() {
+    // TODO: Notify the Agent that all documents were closed.
+    super.tearDown()
+  }
+
+  fun testGetsWorkingGroupLens() {
+    // Do this before starting the edit operation, in order to be subscribed before
+    // the synchronous notification is sent, because it is not buffered or queued anywhere.
+    val foldingRangeFuture = subscribeToTopic(CodyInlineEditActionNotifier.TOPIC_FOLDING_RANGES)
+
+    val editor = myFixture.editor
+    assertFalse(editor.inlayModel.hasBlockElements())
```

Comment:
Hypothetical question, but is an assertion like this *after* a `runInEdtAndWait` likely to assert what you want in the JVM memory model (or at least do it in practice...)